### PR TITLE
Add test vectors for ed25519 edge cases

### PR DESCRIPTION
**Pull request by graydon**
This contains some new test vectors for testing dalek against some known edge cases of ed25519. I'm about to file a corresponding stellar-core change that runs the same test vectors through libsodium (they both behave the same).

### CODE REVIEW

File: soroban-env-host/tests/ed25519_edge_cases.rs [hunk 1]
```diff
@@ -0,0 +1,1015 @@
...
+// of boundary cases for public keys and signatures (14 here being the 8
+// canonical encodings of the 8 small-order points, plus 6 non-canonical
+// encodings of some of these points; and then there's a test vector for such a
+// small-order point occurring in either the public key _or_ the signature, so
+// 14 x 14). We use these test vectors to verify that our implementation
+// _rejects_ them all.
+#[test]
+fn check_zcash_test_vectors_are_rejected() {
+    for (i, test_vector) in ZCASH_TEST_VECTORS.iter().enumerate() {
+        println!("IACR 2020/1244 Test vector {}", i);
```
Comment (leighmcculloch) [inline soroban-env-host/tests/ed25519_edge_cases.rs:73]: This log line references IACR but this test is testing Zcash test vectors.

File: soroban-env-host/tests/ed25519_edge_cases.rs [hunk 2]
```diff
@@ -0,0 +1,1015 @@
...
+        let signature: [u8; SIGNATURE_LENGTH] = hex::decode(test_vector.signature)
+            .unwrap()
+            .try_into()
+            .unwrap();
+        let vk = VerifyingKey::from_bytes(&public_key).unwrap();
+        let sig = Signature::from_bytes(&signature);
+        // The ZIP-215 test vectors are all valid signatures of the message "Zcash",
+        // but we want to reject them because they're all small-order points.
+        const ZIP215_TEST_MESSAGE: &[u8] = b"Zcash";
+        assert!(vk.verify_strict(&ZIP215_TEST_MESSAGE, &sig).is_err());
```
Comment (leighmcculloch) [inline soroban-env-host/tests/ed25519_edge_cases.rs:87]: There'd be value running these test vectors via the Env ed25519 verify host function to ensure that the way the host function uses ed25519-dalek doesn't result in divergence.

**Review by leighmcculloch — APPROVED** [review]
👏🏻 Couple suggestions inline, but ✅ regardless.

File: soroban-env-host/tests/ed25519_edge_cases.rs (hunk 1 above)
Comment (graydon) [reply soroban-env-host/tests/ed25519_edge_cases.rs:73]: Oops, fixed

File: soroban-env-host/tests/ed25519_edge_cases.rs (hunk 2 above)
Comment (graydon) [reply soroban-env-host/tests/ed25519_edge_cases.rs:87]: fair point, added.
